app/src/main/java/ca/bc/gov/bchealth/ui/healthpass/HealthPassViewModel.kt

```diff
@@ -28,14 +28,38 @@ class HealthPassViewModel @Inject constructor(
 
     private val _uiState = MutableStateFlow(HealthPassUiState())
     val uiState: StateFlow<HealthPassUiState> = _uiState.asStateFlow()
-
+    var isAuthenticationRequired: Boolean = true
+    var isBcscLoginRequiredPostBiometrics: Boolean = false
     val healthPasses = repository.patientsVaccineRecord.map { records ->
         records.map { record ->
             record.toUiModel()
         }
     }
 
-    val onBoardingRequired: Flow<Boolean> = onBoardingRepository.onBoardingRequired
+    fun launchCheck() = viewModelScope.launch {
+        when {
+            onBoardingRepository.onBoardingRequired -> {
+                isBcscLoginRequiredPostBiometrics = true
```

Comment:
put it as a separate condition in when statement and ass additional UI state for the same